ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthConfig.java

```diff
@@ -65,23 +67,60 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     long currentTime = System.currentTimeMillis() / 1000;
     ProtocolSpec current = protocolSchedule.getForNextBlockHeader(header, currentTime);
     Optional<ScheduledProtocolSpec> next = protocolSchedule.getNextProtocolSpec(currentTime);
+    Optional<ScheduledProtocolSpec> last = protocolSchedule.getLastProtocolSpec();
 
     ObjectNode result = mapperSupplier.get().createObjectNode();
     ObjectNode currentNode = result.putObject("current");
     generateConfig(currentNode, current);
-    result.put("currentHash", configHash(currentNode));
+    String currentHash = configHash(currentNode);
+    String nextHash = "";
+    String nextNextHash = "";
+    result.put("currentHash", currentHash);
     if (next.isPresent()) {
       ObjectNode nextNode = result.putObject("next");
       generateConfig(nextNode, next.get());
-      result.put("nextHash", configHash(nextNode));
+      nextHash = configHash(nextNode);
+      // get next from next
+      Optional<ScheduledProtocolSpec> nextNext =
+          protocolSchedule.getNextProtocolSpec(next.get().fork().milestone());
+      if (nextNext.isPresent()) {
+        ObjectNode nextNextNode =
+            mapperSupplier.get().createObjectNode(); // don't include in the result
+        generateConfig(nextNextNode, nextNext.get());
+        nextNextHash = configHash(nextNextNode);
+      }
+      result.put("currentForkId", getForkIdHashAsString(currentHash, nextHash));
```

Comment:
Shouldn't this be using genesis block hash and blockNumber/timestamp as inputs, not eth_config hash?

Reply:
yep had to add some wiring to get this in place